app/code/Magento/InventorySales/Model/IsProductSalableForRequestedQtyCondition/IsAnySourceItemInStockCondition.php

```diff
@@ -56,16 +55,16 @@ public function __construct(
      */
     public function execute(string $sku, int $stockId, float $requestedQty): ProductSalableResultInterface
     {
-        $errors = [];
         $isValid = $this->isAnySourceInStockCondition->execute($sku, $stockId);
-        if (!$isValid) {
-            $errors = [
-                $this->productSalabilityErrorFactory->create([
-                    'code' => 'stock_item_is_any_source_in_stock-no_source_items_in_stock',
-                    'message' => __('There are no source items with in stock status')
-                ])
-            ];
+        if ($isValid) {
+            return $this->productSalableResultFactory->create(['errors' => []]);
         }
+        $errors = [
```

Comment:
Better to avoid nested code.
E.g.
```
$data = [
        'code' => 'stock_item_is_any_source_in_stock-no_source_items_in_stock',
        'message' => __('There are no source items with in stock status')
 ];
$errors[] = $this->productSalabilityErrorFactory->create($data);
```